### PR TITLE
Update UI templates and scripts to show the version number at the bottom of the document

### PR DESCRIPTION
**Pull request by dweng0**
Following on from the discussion regarding versioning <https://github.com/autonity/docs.autonity.org/issues/225#issuecomment-2368812592>.  This change adds a version label to the bottom of the contract template used to render contract documentation. 

### Summary of Changes:

- `footer-navigation` has been renamed to `footer` to better reflect its expanded role in managing both footer navigation and versioning.
- Refactored the `footer.lua` so that it accepts version attributes and renders them.
- Added missing chevron icons used by `footer.lua`.
- Added styling to accommodate the new versioning text.

image

### CODE REVIEW

**Review by jay-cousins — COMMENTED** [review]
Looks good but if I try to navigate with the `Previous` - `Next` buttons:

- Error if I click the `Previous` button when viewing the first contract page `Accountability`. The first page either shouldn't have a `Previous` button, or it should go to the Contracts top-level home page which could show an index of all the contract pages?
- Error if I click the `Next` button when viewing the first contract page `Accountability`.

**Comment by szemate** [comment]
> The first page either shouldn't have a Previous button, or it should go to the Contracts top-level home page which could show an index of all the contract pages?

It's a bug in the documentation generator, it does case-sensitive sorting of the contracts and therefore ACU comes before Accountability. I've added a fix to #235

File: reference/contracts/autonity/test.md
Comment (dweng0) [inline reference/contracts/autonity/test.md:1]: Was an example contract that is no longer required

File: reference/contracts/test.md
Comment (dweng0) [inline reference/contracts/test.md:1]: Was an example contract that is no longer requried

File: _filters/contract-footer.lua [hunk 1]
```diff
@@ -26,14 +26,22 @@ function Div(el)
       local next_url = el.attributes["next-url"]
       local next_contract = el.attributes["next-contract"] or "Next contract"
       local next_text = "Next"
+
+      local version = el.attributes["version"] or "latest"
```
Comment (szemate) [inline _filters/contract-footer.lua]: I think it would be better not to show the version if it's missing from the markdown, "latest" can be misleading.

Comment (szemate) [reply _filters/contract-footer.lua]: ... or it could be "unknown"

Comment (dweng0) [reply _filters/contract-footer.lua]: Hmm my reasoning here was that it is the latest known version. But I agree I think using "latest" or "unknown" would be misleading and it makes more sense to not show anything if the version is not provided

**Review by szemate — DISMISSED** [review]

**Review by jay-cousins — DISMISSED** [review]
lgtm

**Comment by dweng0** [comment]
> lgtm

@aiman Was involved in the discussion so i'd like his eyes on it first!

**Comment by aiman** [comment]
Reached out to @dweng0 regarding previewing the changes in this PR

**Comment by dweng0** [comment]
@aiman once pr <https://github.com/autonity/docs.autonity.org/pull/238> has been merged to develop, I'll merge those changes into this branch and you should then be able to preview the changes without the need for a "test.md" file.

**Comment by dweng0** [comment]
@aiman  You should now be able to use `apidocgen` to render the files and test this

**Comment by aiman** [comment]
@szemate Quite a few untracked files, does look right?

```
$ apidocgen
...
$ git status
HEAD detached at origin/update-ui-templates-and-scripts-for-documentation-versioning
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	deleted:    reference/api/accountability/index.md
	deleted:    reference/api/asm/acu/index.md
	deleted:    reference/api/asm/index.md
	deleted:    reference/api/asm/stabilization/index.md
	deleted:    reference/api/asm/supplycontrol/index.md
	deleted:    reference/api/aut/index.md
	deleted:    reference/api/aut/op-prot/index.md
	deleted:    reference/api/liquid-newton/index.md
	deleted:    reference/api/oracle/index.md
	deleted:    reference/api/web3/index.md

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	reference/api/ACU.md
	reference/api/Accountability.md
	reference/api/Autonity.md
	reference/api/Liquid-Newton.md
	reference/api/Oracle.md
	reference/api/Stabilization.md
	reference/api/Supply-Control.md

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by szemate** [comment]
> Quite a few untracked files, does look right?

Yes, it removes the old API documentation from the target directory and then adds the new files.

**Review by aiman — DISMISSED** [review]
This PR LGTM, but we'll need to perform a UI review for all the UI changes thus far that haven't been reviewed.

**Comment by dweng0** [comment]
Closing and reopening the pr to re trigger checks

**Review by szemate — APPROVED** [review]

**Comment by aiman** [comment]
The "build" status check is simply not returning a status, so Github cannot determine if the branch is good to merge or not.
Screenshot from 2024-10-02 17-27-37

This is the code in this branch:
https://github.com/autonity/docs.autonity.org/blob/update-ui-templates-and-scripts-for-documentation-versioning/.github/workflows/pr.yml#L7

This `build` test was passing in master, see recent commits:
https://github.com/autonity/docs.autonity.org/commits/master/

**Comment by dweng0** [comment]
@aiman  This is very odd. There's no GitHub action logo and the action that is failing doesn't have a `details` link.

I checked out the branch and copied it over to the [engineering sandbox](https://github.com/clearmatics/engineering-sandbox/pull/14) and added the two checks above into the branch rules:

image

They show correctly with a github action "details" link for both:
image

I'm not entirely sure why it is not working for this branch but it could be that the rule needs to be removed and re added again?

**Review by aiman — APPROVED** [review]

**Comment by aiman** [comment]
LGTM

**Pull request merged**